src/axolotl/utils/data.py

```diff
@@ -440,7 +440,7 @@ def load_prepare_datasets(
     split="train",
 ) -> Tuple[Dataset, Dataset, List[Prompter]]:
     dataset, prompters = load_tokenized_prepared_datasets(
-        tokenizer, cfg, default_dataset_prepared_path
+        tokenizer, cfg, default_dataset_prepared_path, split=split
```

Comment:
let's hold off on handling the split kwarg here. It currently gets manually handled below and I'd prefer to clean this part up in a separate PR.

Reply:
Are you sure that works? Because inside `load_tokenized_prepared_datasets` you pick the dataset array like this:

```py
cfg_datasets = cfg.test_datasets if split == "test" else cfg.datasets
```

If you never set the split, then it will always default to `train` and therefore `cfg_datasets` will never be `cfg.test_datasets`.

Reply:
`split` might be a bit of bad naming, since it's an argument to `load_tokenized_prepared_datasets` to reuse the same logic for the test_datasets and datasets